network/topology.go

```diff
@@ -0,0 +1,48 @@
+package network
+
+import (
+	"fmt"
+	"sort"
+	"sync"
+	"sync/atomic"
+	"time"
+)
+
+type Topology interface {
+	MakeConnections(agents map[int]agentContainer)
+}
+
+type LinearTopology struct{}
+
+func (t LinearTopology) MakeConnections(agents map[int]agentContainer) {
+	keys := make([]int, 0)
+	for k := range agents {
+		keys = append(keys, k)
+	}
+	sort.Ints(keys)
+
+	startTime := time.Now()
+	wg := sync.WaitGroup{}
+	wg.Add(len(agents))
+	cntAgentsConnected := int64(0)
+
+	for i, key := range keys {
+		if i <= len(keys)-1 {
+			go func(i int) {
+				defer wg.Done()
+
+				source := agents[key].agent
```

Comment:
Because this is in a go routine and variable key is changed from outside, I assume this should be `source := agents[keys[i]].agent`